### DISCUSSION TITLE
Does the value saved in /var/cache/cups/[printer_name].strings meet the expectations?

### DISCUSSION THREAD

**Discussion by xiaowen581** [OpenPrinting/cups#991] [Q&A] [answered]
The strings returned from the printer's printer-strings-uri is
`
"media-type.com.epson-thickpaper1" = "Thick paper1";
`
However, after installing the printer with everywhere, the strings saved in /var/cache/cups/[printer_name].strings is 
`
"media-type.com.epson-thickpaper-1" = "Com.epsonThickpaper1";
`

Therefore, when using cupsLocalizeDestValue(xxx, xxx, xxx, "media-type", "com.epson-thickpaper-1"), it returns "Com.epsonThickpaper1". However, I think that "Thick paper1" should be the correct answer, right?

**System Information:**
 - OS and its version: Ubuntu 22.04(WSL)
 - CUPS version: 2.4.10
 
the ppd created by everywhere :
[epson.ppd.txt](https://github.com/user-attachments/files/15936482/epson.ppd.txt)
the strings cache :
[epson.strings.txt](https://github.com/user-attachments/files/15936519/epson.strings.txt)

**Comment by xiaowen581** [comment]
I tried to change the line in _ppdCreateFromIPP2 
from 
`
cupsFilePrintf(fp, "*MediaType %s: \"<</MediaType(%s)>>setpagedevice\"\n", ppdname, ppdname);
`
to
`
cupsFilePrintf(fp, "*MediaType %s/%s: \"<</MediaType(%s)>>setpagedevice\"\n", ppdname, msgstr, ppdname);
`
then I get my expected outcome from cupsLocalizeDestValue.

For 2.4.10, It seems that the choice->text in the _ppdCacheCreateWithPPD doesn't obtain the message in C.MediaType cause the issue?

**Comment by xiaowen581** [comment]
I think this issue could be a duplicate of #546

**Reply by michaelrsweet** [reply]
Actually, #86 which is resolved for CUPS 2.5 (not a whole lot we can do for 2.4.x unfortunately...)

**Comment by michaelrsweet** [comment, marked as answer]
There are known localization issues in CUPS 2.4 that are resolved in CUPS 2.5. Unfortunately, the scope of the changes is more than we'd want to include in a patch update so it will have to wait for 2.5.0 later this year.
